File: src/python/asset-manipulations/asset-conversion/hbd_to_hive.py
from wax import create_wax_foundation


# Create a Wax Foundation instance
wax = create_wax_foundation()

# Assume these amounts represent the HBD, base, and quote
hbd_amount = 1_000
base_amount = 1_500
quote_amount = 2_000

# Convert amounts to `NaiAsset`
hbd_asset = wax.hbd.satoshis(hbd_amount)
base_asset = wax.hbd.satoshis(base_amount)
quote_asset = wax.hive.satoshis(quote_amount)

# Use `hbd_to_hive` to perform the conversion
hive_asset = wax.hbd_to_hive(hbd_asset, base_asset, quote_asset)
print(f"Converted Hive Asset: {hive_asset}")

File: src/rust/asset_manipulations/asset_convertions/hbd_to_hive.rs
use wax::prelude::*;

fn main() -> Result<(), Box<dyn std::error::Error>> {
    // Create a Wax Foundation instance
    let wax_api = create_wax_foundation(None);

    // Assume these amounts represent the HBD, base, and quote
    let hbd_amount = 1_000;
    let base_amount = 1_500;
    let quote_amount = 2_000;

    // Convert amounts to `NaiAsset`
    let hbd_asset = wax_api.hbd_satoshis(hbd_amount)?;
    let base_asset = wax_api.hbd_satoshis(base_amount)?;
    let quote_asset = wax_api.hive_satoshis(quote_amount)?;

    // Use `hbd_to_hive` to perform the conversion
    let hive_asset = wax_api.hbd_to_hive(&hbd_asset, &base_asset, &quote_asset)?;
    println!("Converted Hive Asset: {hive_asset:?}");

    Ok(())
}
